Library/Homebrew/dev-cmd/tests.rb

```diff
@@ -38,6 +38,13 @@ def tests_args
   def tests
     tests_args.parse
 
+    consider_patchelf = OS.linux? && (HOMEBREW_PATCHELF_RB || HOMEBREW_READELF_RB)
+    if consider_patchelf
+      require "bundler/cli"
+      # :: PATCHELF_VERSION
+      Bundler::CLI.start ["add", "patchelf", "--version", "~> 1.0.0", "--skip-install"]
```

Comment:
i don't particularly like this, 
approach, but LOAD_PATH had problem loading a recursive dep,
did this as a fix that guarantees to work, but may leave a Gemfile in unnecessary state 
if tests are interrupt  by user. 
what to do :/

Reply:
@rmNULL Can we take a step back: without this, if you just have `Homebrew.install_gem! "patchelf"` 1) where needed in the code or 2) in `spec_helper.rb`: what are the errors you see?

Reply:
No problemo.
Did a hard reset, and added a conditional to check for readelf flags only on Linux.

The old problem still exists on a smaller level, even after `Homebrew.install_gem!`
it fails to require 'bindata/version', 'patchelf/exceptions' on 6 cases( when running `brew tests`)
This is odd considering  those 6 failing specs run correct when passed to  `brew test --only `.

Reply:
I think if `brew tests` is failing only when the necessary variable is being set then this can be merged without being fixed.

Reply:
brew tests only fail when `HOMEBREW_READELF_RB` is set in the env(the CI tests speaks for that i guess).